modules/ROOT/pages/type-definitions/directives/genai.adoc

```diff
@@ -0,0 +1,170 @@
+This embedding is then compared to the node vector embeddings in the database and the result set 
+
+[NOTE] 
+.Prerequisites
+==== 
+* The node vector embeddings already exist
+* The embeddings must have been created using the same model
+* Queries by vector index cannot be performed across multiple labels
+* Queries by phrase require credentials for the Neo4j GenAI plugin
+====
+
+== `@vector`
+
+Use an existing vector embedding on a node in the database to perform a nearest neighbor search.
+
+=== Definition
+
+[source, graphql]
+----
+"""
+Informs @neo4j/graphql that there is a vector index in the database which allows users to search by the index in the generated schema.
+"""
+directive @vector(indexes: [VectorInput]!) on OBJECT
+----
+
+`VectorInput` is defined as follows:
+
+[source, graphql]
+----
+input VectorInput {
+  name: String!
+  dimension: Int!
+  similarityFunction: 'cosine' | 'euclidean'
+  property: String!
+}
+----
+
+=== Usage
+
+.Type definition
+[source, graphql]
+----
+type Product @vector(
+  name: "Interesting",
+  dimension: 1536,
+  similarityFunction: 'cosine'
+  property: "embeddingPropertyName"
+) {
+  name: String!
+  description: String!
+}
+----
+
+Adding the vector directive to the `Product` type creates a a top-level query named `productsInterestingVector`.
+The name is a combination of the plural form of the type name, the index name and `Vector`:
+
+.Query schema
+[source, graphql]
+----
+type Query {
+    productsInterestingVector(text: String!, where: ProductScoreWhere, sort: [ProductVectorSort!], limit: Int, offset: Int): [ProductVectorResult!]!
+    products(where: ProdcutOperationWhere): ProductOperation!
+}
+type ProductOperation {
+    connection(first: Int, after: String, sort: [ProductConnectionSort!]): ProductConnection!
+    aggregation: ProductAggregation!
+}
+type ProductConnection {
+    pageInfo: PageInfo! # Required by relay
+    edges: [ProductEdge!]! # Required by relay
+}
+type ProductEdge {
+    cursor: String! # Required by relay
+    node: Product! # Required by relay
+    score: Float
+}
+type Product {
+    name: String!
+    description: String!
+}
+----
+
+.Example query
+[source, graphql]
+----
+query {
+  productsInterestingVector(
+    text: "This is the submitted text which is converted into a vector embedding by the GenAI plugin",
+    where: { edges: { score: { gt: 0.5 } } }
+    sort: [{ product: { name: ASC } }]
+  ) {
+    connection {
+      edges {
+        score
+        node {
+          name
+        }
+      }
+    }
+  }
+}
+----
+
+
+== `@genAI`
+
+Perform a query which utilizes the link::https://neo4j.com/docs/cypher-manual/current/genai-integrations/[Neo4j GenAI plugin] to create a vector embedding for a search phrase and then compares it to existing vector embeddings on nodes in the database.
+Requires credentials for the plugin.
```

Comment:
Should this become a note or is it not so crucial that needs to be highlighted?

Reply:
i thought about this - it's in an admonition box higher up ("prerequisites"), but i can absolutely add another admonition here